src/algorithm/retrieveWebidTrustedOidcIssuers.ts

```diff
@@ -1,48 +1,61 @@
-import { DataFactory } from "n3";
-import rdfDereferencer from "rdf-dereference";
-import type { Quad, Stream } from "rdf-js";
+// eslint-disable-next-line no-shadow
+import { URL } from "url";
+import type { Quad } from "@rdfjs/types";
+import { DataFactory, Parser, Store } from "n3";
+// eslint-disable-next-line no-shadow
+import fetch from "node-fetch";
 import { WebidDereferencingError } from "../error/WebidDereferencingError";
+import { WebidIriError } from "../error/WebidIriError";
 import { WebidParsingError } from "../error/WebidParsingError";
 import type { RetrieveOidcIssuersFunction } from "../type";
 
-const defaultGraph = DataFactory.defaultGraph();
-const oidcIssuer = DataFactory.namedNode(
-  "http://www.w3.org/ns/solid/terms#oidcIssuer"
-);
-
-async function dereferenceWebid(webid: string): Promise<Stream<Quad>> {
+async function dereferenceWebid(webid: string): Promise<string> {
   try {
-    return (await rdfDereferencer.dereference(webid)).quads;
+    const response = await fetch(webid, {
+      headers: {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        Accept: "text/turtle",
+      },
+    });
+    return await response.text();
   } catch (e: unknown) {
     throw new WebidDereferencingError(webid);
   }
 }
 
+function parseRdf(rdf: string, baseIRI: string): Store<Quad> {
+  try {
+    const store = new Store();
+    store.addQuads(new Parser({ baseIRI }).parse(rdf));
+    return store;
+  } catch (e: unknown) {
+    throw new WebidParsingError();
+  }
+}
+
 export async function retrieveWebidTrustedOidcIssuers(
   webid: string,
   getIssuers?: RetrieveOidcIssuersFunction
 ): ReturnType<RetrieveOidcIssuersFunction> {
+  let webidUrl: URL;
+  try {
+    webidUrl = new URL(webid);
+  } catch (e: unknown) {
+    throw new WebidIriError(webid);
+  }
   if (typeof getIssuers !== "undefined" && getIssuers !== null) {
     return getIssuers(webid);
   }
+  const store = parseRdf(
+    await dereferenceWebid(webid),
+    webidUrl.toString().substring(0, webid.indexOf("#"))
+  );
+  const quads = store.getQuads(
```

Comment:
N3.js has a `getObjects` function which would be useful here.

Reply:
Yes! :)
Fixed in https://github.com/solid/access-token-verifier/pull/24/commits/57769c93f97c83a21edf089e4dc53aee3c1751c6